### ISSUE TITLE
rocket::http::Status as oapi responder

### ISSUE THREAD

**Issue by l4l** [GREsau/okapi#20]
It seems that `rocket::http::Status` does not implement `OpenApiResponder`, so that simple route will fail:

```rust
use rocket::http::Status;

#[get("/")]
fn func() -> Status { Status::Ok }
```

It's quite tricky, because the interface is not supposed to have access to the object itself, only type. As a temporary workaround https://github.com/l4l/okapi/commit/9206ab23b19a9aff57c7cd02fb8ea588d7810057 may work, though I don't really think it's a good idea to merge it.

Maybe some wrapper should be considered for statuses? Procedural macros cannot properly determine whether the item is particular type or not in general, afaik, although they may emit warnings for a wrapper hinting.

**Comment by megametres** [comment]
I think that it is safe to assume that if a route return's type is a rocket::http::Status when everything goes as expected the returned status will be Status::Ok. If not, I think that implementing something like #[okapi(code = "404")] would be the best course of action, altough I do not have a clue if it would be hard doing so...

**Comment by GREsau** [comment]
Related: https://github.com/GREsau/okapi/issues/15#issuecomment-604645426

**Comment by ralpha** [comment]
This has been fixed in 1740ebc59ff1135ea4e0c4bc73a14cf99e5e8413.
Okapi now uses the "default" setting in OpenAPI: https://spec.openapis.org/oas/v3.0.0#fixed-fields-13
![Screenshot from 2021-09-09 18-19-27](https://user-images.githubusercontent.com/2608639/132724044-241cc92b-fc97-465d-a23b-e10ab0081d1f.png)
![Screenshot from 2021-09-09 18-21-23](https://user-images.githubusercontent.com/2608639/132724365-c71d3a84-0d75-41b0-96ff-3c61df063825.png)

This is a screenshot of the example: https://github.com/GREsau/okapi/blob/master/examples/special-types/src/main.rs#L17
It also works with just the `rocket::http::Status` (or other combination you might expect, like `Result<Json<()>, rocket::http::Status>`)
The documentation for these endpoint might be a bit scarce, because there is not much info that can be displayed.
